src/host/inputBuffer.cpp

```diff
@@ -816,6 +816,26 @@ void InputBuffer::_HandleTerminalInputCallback(const TerminalInput::StringType&
 
         for (const auto& wch : text)
         {
+            if (wch == UNICODE_NULL)
+            {
+                // Convert null byte back to input event with proper control state
+                const auto zeroKey = OneCoreSafeVkKeyScanW(0);
+                uint32_t ctrlState = 0;
+                if (WI_IsFlagSet(zeroKey, 0x100))
+                {
+                    ctrlState |= SHIFT_PRESSED;
+                }

```

Comment:
Just a nit, but I recently discovered there's `WI_SetFlagIf` macro that makes this kind of code a little neater. You can compress these conditions into one line each with something like:

```
WI_SetFlagIf(ctrlState, SHIFT_PRESSED, WI_IsFlagSet(zeroKey, 0x100));
```

It only works with single flags though, so I think you'd have to use `LEFT_CTRL_PRESSED` and `LEFT_ALT_PRESSED` for the other two, but that should still work. It's not important though. Just an idea.

Reply:
@j4james Thanks. It's much cleaner. It works on my computer.